modules/network_synchronizer/net_utilities.h

```diff
@@ -199,7 +196,7 @@ T StatisticalRingBuffer<T>::average() const {
 	}
 	a = a / T(data.size());
 	T b = avg_sum / T(data.size());
-	ERR_FAIL_COND_V_MSG(ABS(a - b) > (CMP_EPSILON * 4.0), b, "The `avg_sum` accumulated a sensible precision loss: " + rtos(ABS(a - b)));
+	ERR_FAIL_COND_V_MSG(a - b > (CMP_EPSILON * 4.0), b, "The `avg_sum` accumulated a sensible precision loss: " + rtos(a - b));
```

Comment:
`a - b` may be negative, and in that case it invalidates the check. While `ABS` helps to get just the delta. Is this wrong?

Reply:
We are using unsigned int, so this is a bug, we need to cast the type instead to remove `ABS`. 
Can you please use:
```cpp
	int64 a = data[0];
	for (uint32_t i = 1; i < data.size(); i += 1) {
		a += data[i];
	}
	a = a / int64(data.size());
	int64 b = int64(avg_sum) / int64(data.size());
	ERR_FAIL_COND_V_MSG(ABS(a - b) > (CMP_EPSILON * 4.0), b, "The `avg_sum` accumulated a sensible precision loss: " + rtos(ABS(a - b)));
```
Excellent catch!